aten/src/ATen/native/DistributionTemplates.h

```diff
@@ -218,6 +218,11 @@ Tensor& normal_out_impl(Tensor& output, const Tensor& mean, double std, c10::opt
 template<template<typename> class normal_kernel, typename RNG>
 Tensor& normal_out_impl(Tensor& output, double mean, const Tensor& std, c10::optional<Generator> gen) {
   TORCH_CHECK(!std.is_complex(), "normal expects standard deviation to be non-complex");
+  if (std.device().type() == c10::DeviceType::CPU) {
```

Comment:
I think it's ok to do this for cuda also. 
Ideally, we would change the implementation of normal_impl_ to check the values of std during generation (that hopefully would have a small perf impact), but since we are not doing that neither for cpu nor cuda, it's ok to have thin `min` check for both devices.

Reply:
This will cause a sync, right? If so, we should add a note in the documentation of torch.normal that when constructing a tensor on a cuda device it synchronizes that device with the CPU (analogous to notes we have for linear algebra operations).

Reply:
I think you're right, in order for the `.item<bool>()` call to work, I'm pretty sure it has to be doing a sync. I've added a note in the docs